apps/s_server.c

```diff
@@ -3265,7 +3265,7 @@ static int www_body(int s, int stype, int prot, unsigned char *context)
             BIO_printf(io, "---\n");
             print_stats(io, SSL_get_SSL_CTX(con));
             BIO_printf(io, "---\n");
-            peer = SSL_get_peer_certificate(con);
+            peer = SSL_get1_peer_certificate(con);
```

Comment:
```suggestion
            peer = SSL_get0_peer_certificate(con);
```
and then delete `X509_free(peer)` below.